docs/02-guides/override-a-component.mdx

```diff
@@ -2,14 +2,177 @@
+
+```
+my-project
+└── app
+    └── theme
+```
+
+That’s it! You have successfully created the folder that will be the root of
+your custom theme !
+
+### In your own extension
+
+You can also extend theme in your extension, please read
+[Register an extension](/docs/remixed/guides/register-an-extension) guide to
+know how to create your own extension.
+
+First in your extension definition file, add the `theme` entry :
+
+```javascript title="extension/my-extension/index.ts"
+import path from "path";
+import { defineExtension } from "@front-commerce/core";
+
+export default function themeChocolatine() {
+  return defineExtension({
+    name: "MyExtension",
+    // highlight-next-line
+    theme: path.join(__dirname, "theme"),
+    configuration: {
+      providers: [],
+    },
+  });
+}
+```
+
+And now theme file will be injected in your project !
+
+## Override a component
+
+Let's add the description of a `Product` to a `ProductItem` as an example of
+overriding the base theme.
+
+The original file is:
+[`node_modules/@front-commerce/theme-chocolatine/theme/modules/ProductView/ProductItem/ProductItem.js`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/theme-chocolatine/theme/modules/ProductView/ProductItem/ProductItem.jsx)
+
+:::info
+
+Please refer to the
+[Front-Commerce’s folder structure documentation page](/docs/2.x/concepts/front-commerce-folder-structure)
+to get a better understanding of how components are organized in the base theme.
+
+:::
+
+1. copy it to:
+   `extension/my-extension/theme/ProductView/ProductItem/ProductItem.jsx` or
+   `app/theme/ProductView/ProductItem/ProductItem.jsx` (depending on the
+   [override method you've decided to use](#creating-the-theme-folder))
+2. add the description after the `<ProductOverview />` component in your
+   `ProductItem` with `{props.description}`.
+
+**But you are not done yet!** The `description` information is not included in
+the GraphQL fields fetched by the application in the base theme. You will thus
+need to update the fragment related to `ProductItem`.
+
+The original fragment file is collocated with the original component at:
+
+1. copy the
+   [`original ProductItemFragment`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/main/packages/theme-chocolatine/theme/modules/ProductView/ProductItem/ProductItemFragment.gql?ref_type=heads)
+   to the equivalent location in your extension.
+1. add the field `description` to the fragment.
+
+   ```diff title="app/theme/modules/ProductView/ProductItem/ProductItemFragment.gql"
+   fragment ProductItemFragment on Product {
+     imageUrl
+   + description
+     ...ProductOverviewFragment
+     ...ProductItemActionsFragment
+   }
+   ```
+
+1. restart the application
+
+The data will now be fetched from GraphQL every time the `ProductItem` is used
+(product listings, upsells…) and will be available for you to render it as
+wanted.
+
+### Reuse the original component
```

Comment:
**thought:** I'm not really sure to see the use case for this... It feels like overriding the component directly should even cover the example here?

Reply:
There are some use cases where you would want to use the original component, for example, if I only want to extend the original component without overriding i would do

```tsx
// app/theme/modules/OriginalComponent

import OriginalComponent from "@front-commerce/theme/modules/OriginalComponent"

const ExtendedComponent = () => {
	return (
		<Stack>
			<h1>My Extended Component</h1>
			<OriginalComponent />
		</Stack>
	)
}